tools/render-test/shader-input-layout.cpp

```diff
@@ -698,42 +698,182 @@ namespace renderer_test
+    /* static */SlangResult ShaderInputLayout::writeBinding(BindRoot* bindRoot, const ShaderInputLayoutEntry& entry, const void* data, size_t sizeInBytes, WriterHelper writer)
     {
-        FILE * f = fopen(fileName.getBuffer(), "wb");
-        if (!f)
+        typedef slang::TypeReflection::ScalarType ScalarType;
+
+        slang::TypeReflection::ScalarType scalarType = slang::TypeReflection::ScalarType::None;
+
+        slang::TypeLayoutReflection* typeLayout = nullptr;
+
+        if (bindRoot && entry.name.getLength())
         {
-            return SLANG_FAIL;
+            BindLocation location;
+            if (SLANG_SUCCEEDED(bindRoot->parse(entry.name, "", writer, location)))
+            {
+                // We should have the type of the item
+                typeLayout = location.m_typeLayout;
+            }
         }
 
-        const auto& entries = layout.entries;
+        slang::TypeLayoutReflection* elementTypeLayout = nullptr;
 
-        for (int i = 0; i < entries.getCount(); ++i)
+        if (typeLayout)
         {
-            const auto& entry = entries[i];
-            if (entry.isOutput)
+            switch (typeLayout->getKind())
             {
-                BindSet::Value* buffer = buffers[i];
+                
+                //case slang::TypeReflection::Kind::Struct:
+                case slang::TypeReflection::Kind::Array:
+                case slang::TypeReflection::Kind::Matrix:
+                case slang::TypeReflection::Kind::Vector:
+                {
+                    elementTypeLayout = typeLayout->getElementTypeLayout();
+                    break;
+                }
+                case slang::TypeReflection::Kind::Scalar:
+                {
+                    elementTypeLayout = typeLayout;
+                    break;
+                }
+                case slang::TypeReflection::Kind::Resource:
+                {
+                    elementTypeLayout = typeLayout->getElementTypeLayout();
+                    break;
+                }   
+                case slang::TypeReflection::Kind::TextureBuffer:
+                case slang::TypeReflection::Kind::ShaderStorageBuffer:
+                {
+                    elementTypeLayout = typeLayout->getElementTypeLayout();
+                    break;
+                }
+            }
+        }
 
-                unsigned int* ptr = (unsigned int*)buffer->m_data;
+        if (elementTypeLayout)
+        {
+            scalarType = elementTypeLayout->getScalarType();
+        }
 
-                const int size = int(entry.bufferData.getCount());
-                // Must be the same size or less than allocated buffer
-                SLANG_ASSERT(size * sizeof(unsigned int) <= buffer->m_sizeInBytes);
 
-                for (int i = 0; i < size; ++i)
+        switch (scalarType)
+        {
+            // TODO(JS):
+            // Bool is here, because it's not clear across APIs how bool is laid out in memory
```

Comment:
One random thought is that we might want to output these values canonically as hex, and then include a "humane" version of the value as a comment, e.g.:

```
0x3f800000 // 1.0 (float32)
```

Ideally we would then spruce up the comparison logic so that for these tests we use a comparsion that ignores the `//` and the text that follows it, so that the canonical test output is just the hex values, to account for potential differences in printing across OSes (I guess that still doesn't deal with differences across platforms in the actual bits of the result...)